lib/membrane_mp4/muxer/isom.ex

```diff
@@ -0,0 +1,196 @@
+defmodule Membrane.MP4.Muxer.ISOM do
+  @moduledoc """
+  Puts payloaded streams into an MPEG-4 container.
+  """
+
+  # Due to the structure of MPEG-4 containers, it is not possible
+  # to send buffers with `mdat` box content right after they are
+  # processed — we need to know the size of this box up front.
+  # The current solution requires storing all incoming buffers
+  # in memory until the last track ends.
+  # Once some kind of seeking mechanism is implemented, it will
+  # be possible to send chunks of samples on demand, only
+  # correcting the mdat header after all media data has been sent.
+
+  use Membrane.Filter
+
+  alias Membrane.{Buffer, Time}
+  alias Membrane.MP4.Container
+  alias Membrane.MP4.Muxer.{MovieBox, Track}
+
+  @ftyp [
+    ftyp: %{
+      children: [],
+      fields: %{
+        compatible_brands: ["isom", "iso2", "avc1", "mp41"],
+        major_brand: "isom",
+        major_brand_version: 512
+      }
+    }
+  ]
+
+  @ftyp_size @ftyp |> Container.serialize!() |> byte_size()
+  @mdat_data_offset 8
+
+  def_input_pad :input,
+    demand_unit: :buffers,
+    caps: Membrane.MP4.Payload,
+    availability: :on_request
+
+  def_output_pad :output, caps: :buffers
+
+  def_options fast_start: [
+                spec: boolean,
+                default: false,
+                description: """
+                Whether the container metadata (`moov` atom) should be placed before media data.
+
+                Especially suitable for streaming, where makes it possible to start playback before all
+                media data arrives. Equivalent of `ffmpeg`'s `-movflags faststart` option.
```

Comment:
It sounds familiar ;) However, as in general we should take the approach of making our framework more user friendly, I'd focus on the effect rather than technical details of how it is achieved (but not omit that). Simply put -swap the order of sentences and put `makes it possible...` first and then explain atoms etc.